core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -1521,6 +1521,13 @@ public Builder setPreviousFileLocation(String previousFileLocation) {
       return this;
     }
 
+    public Builder setLastUpdatedMillisIfNull(long lastUpdatedAt) {
+      if (this.lastUpdatedMillis == null) {
```

Comment:
just my thought, its a bit strange to have a builder silently ignore an argument.  i guess its from: https://github.com/apache/iceberg/pull/14380#discussion_r2449280988 ?

It would make more sense to assert (or perhaps check precondition) , as i think the comment suggested ?